core/src/impl/Kokkos_Default_Graph_Impl.hpp

```diff
@@ -80,9 +80,9 @@ struct GraphImpl : private ExecutionSpaceInstanceStorage<ExecutionSpace> {
   // <editor-fold desc="required customizations"> {{{2
 
   template <class NodeImpl>
-  //  requires NodeImplPtr is a shared_ptr to specialization of GraphNodeImpl
-  void add_node(std::shared_ptr<NodeImpl> const& arg_node_ptr) {
-    static_assert(NodeImpl::kernel_type::Policy::is_graph_kernel::value);
+  std::enable_if_t<
+      Kokkos::Impl::is_specialization_of_v<NodeImpl, GraphNodeImpl>>
+  add_node(std::shared_ptr<NodeImpl> const& arg_node_ptr) {
```

Comment:
hmm ... can we give an easy to read error message to our users? Do you really need SFINAE here?

Reply:
SFINAE is not 100% needed now, but a following PR will introduce a new type of node, for which we'll need to specialize `add_node`;

Reply:
ah ok. well, if users never hit this I guess it is fine.

Reply:
@JBludau i updated the PR description. If it's OK for you, please close this thread :wink: